AberrantSMPP/ASyncSocketClient.cs

```diff
@@ -269,6 +274,14 @@ public void Connect(String address, UInt16 port)
 				_TcpClient.NoDelay = true;
 				//if the connection is dropped, drop all associated data
 				_TcpClient.LingerState = new LingerOption(false, 0);
+
+				if(_UseSsl)
+				{
+					_SslStream = new SslStream(_TcpClient.GetStream());
+
+					if (!_SslStream.IsAuthenticated)
+						_SslStream.AuthenticateAsClient(_ServerAddress);
```

Comment:
This makes sense only if we allow for SSL-based client authentication using X509 Certificates. Do you expose properties or methods as to allow/setup such authentication?

Reply:
This line is required, if omitted, the following exception  will be thrown:
_System.InvalidOperationException: This operation is only allowed using a successfully authenticated context._